doc/tutorial/read_mapper/read_mapper_step5.cpp

```diff
@@ -0,0 +1,148 @@
+#include <seqan3/alignment/configuration/all.hpp>
+#include <seqan3/alignment/pairwise/align_pairwise.hpp>
+#include <seqan3/argument_parser/all.hpp>
+#include <seqan3/io/alignment_file/output.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/get.hpp>
+#include <seqan3/search/algorithm/all.hpp>
+#include <range/v3/view/zip.hpp>
+
+using namespace seqan3;
+
+struct reference_storage_t
+{
+    concatenated_sequences<std::string> ids;
+    concatenated_sequences<std::vector<dna5>> sequences;
+};
+
+struct query_storage_t
+{
+    concatenated_sequences<std::string> ids;
+    concatenated_sequences<std::vector<dna5>> sequences;
+    concatenated_sequences<std::vector<phred42>> qualities;
+};
+
+reference_storage_t reference_storage{};
+query_storage_t query_storage{};
+
+void read_input(std::filesystem::path const & reference_path, std::filesystem::path const & query_path)
+{
+    sequence_file_input reference_in{reference_path};
+    reference_storage.ids       = std::move(get<field::ID>(reference_in));
+    reference_storage.sequences = std::move(get<field::SEQ>(reference_in));
+
+    sequence_file_input query_in{query_path};
+    query_storage.ids       = std::move(get<field::ID>(query_in));
+    query_storage.sequences = std::move(get<field::SEQ>(query_in));
+    query_storage.qualities = std::move(get<field::QUAL>(query_in));
+}
+
+void map(uint8_t errors, std::filesystem::path const & sam_path)
+{
+    bi_fm_index index{reference_storage.sequences};
+    alignment_file_output sam_out{sam_path};
```

Comment:
```suggestion
    alignment_file_output sam_out{sam_path, fields<field::ID, ...>};
```

and then you do not need a record_type but just do:

```
sam_out.emplace_back(id, ...);
```